test/jb_splits_store/set.test.js

```diff
@@ -0,0 +1,278 @@
+
+    let mockJbOperatorStore = await deployMockContract(deployer, jbOperatorStore.abi);
+    let mockJbProjects = await deployMockContract(deployer, jbProjects.abi);
+    let mockJbDirectory = await deployMockContract(deployer, jbDirectory.abi);
+
+    await mockJbOperatorStore.mock.hasPermission
+      .withArgs(projectOwner.address, projectOwner.address, PROJECT_ID, SET_SPLITS_PERMISSION_INDEX)
+      .returns(true);
+
+    await mockJbProjects.mock.ownerOf
+      .withArgs(PROJECT_ID)
+      .returns(projectOwner.address);
+
+    await mockJbDirectory.mock.controllerOf
+      .withArgs(PROJECT_ID)
+      .returns(projectOwner.address);
+
+    let jbSplitsStoreFact = await ethers.getContractFactory('JBSplitsStore');
+    let jbSplitsStore = await jbSplitsStoreFact.deploy(
+      mockJbOperatorStore.address,
+      mockJbProjects.address,
+      mockJbDirectory.address
+    );
+
+    let splits = makeSplits(addrs[0].address);
+
+    return { deployer, projectOwner, addrs, jbSplitsStore, splits, mockJbOperatorStore, mockJbProjects, mockJbDirectory };
+  }
+
+  function makeSplits(beneficiaryAddress, count=4) {
+    let splits = []
+    for (let i = 0; i < count; i++) {
+      splits.push({
+        preferClaimed: false,
+        percent: Math.floor(10000000 / count),
+        lockedUntil: 0,
+        beneficiary: beneficiaryAddress,
+        allocator: ethers.constants.AddressZero,
+        projectId: 0
+      });
+    }
+    return splits;
+  }
+
+  it('set(...) and corresponding events', async function () {
+    const { projectOwner, addrs, jbSplitsStore, splits, mockJbOperatorStore, mockJbDirectory } = await setup();
+
+    await mockJbOperatorStore.mock.hasPermission.returns(false);
+    await mockJbDirectory.mock.controllerOf.returns(addrs[0].address);
+
+    const tx = await jbSplitsStore.connect(projectOwner).set(
+      PROJECT_ID,
+      DOMAIN,
+      GROUP,
+      splits
+    );
+
+    // Expect one event per split in splits[]
+    await Promise.all(
+      splits.map(async (split, _) => {
+        await expect(tx)
+          .to.emit(jbSplitsStore, 'SetSplit')
+          .withArgs(PROJECT_ID, DOMAIN, GROUP, Object.values(split), projectOwner.address);
+      }),
+    );
+
+    // Get the current splits (for this proj/dom/group)
+    let splitsStored = await jbSplitsStore.splitsOf(PROJECT_ID, DOMAIN, GROUP);
+
+    //compare every currently stored splits to the one we've just sent
+    for (let [idx, split] of splitsStored) {
+      for (let splitKey of Object.keys(split)) {
+        expect(split.splitKey).to.equal(splits[idx].splitKey);
+      }
+    }
+  })
+
+  it('Create and overwrite existing splits with same ID/Domain/Group', async function () {
+    const { projectOwner, addrs, jbSplitsStore, splits } = await setup();
+
+    await jbSplitsStore.connect(projectOwner).set(
+      PROJECT_ID,
+      DOMAIN,
+      GROUP,
+      splits
+    );
+
+    // 4 new ones, with a new beneficiary for each,
```

Comment:
The comments with "4" no longer make sense with the new makeSplits() count default param. Please update.

Reply:
"Using the default parameter of 4" ?

Reply:
Removed this comment altogether as code is self-explanatory here